packages/smooth_app/lib/temp/product_preferences.dart

```diff
@@ -12,12 +12,57 @@ import 'package:openfoodfacts/model/AttributeGroup.dart';
 import 'package:openfoodfacts/model/Product.dart';
 
 // Project imports:
-import 'package:smooth_app/temp/user_preferences.dart';
 import 'package:smooth_app/temp/preference_importance.dart';
 
-class UserPreferencesModel extends ChangeNotifier {
-  UserPreferencesModel._();
+/// Where we store all product attribute groups and importance.
+///
+/// At init time we need to load from json the localized reference data
+/// for importance and attribute groups.
+/// We also need to store the app or user preferences regarding importance,
+/// e.g. 'nutriscore' is 'very_important' and 'ecoscore' is 'mandatory',
+/// but we cannot decide here how to store those preferences and rather let
+/// the developer implement [saveImportance] and [loadImportance]
+class ProductPreferences extends ChangeNotifier {
+  ProductPreferences(this.saveImportance, this.loadImportance);
 
+  /// Saves the importance of an attribute, e.g. in a SharedPreferences.
+  ///
+  /// Ex:
+  /// ```dart
+  /// (String attributeId, int importanceIndex) async =>
+  ///     await mySharedPreferences.setInt(attributeId, importanceIndex);
+  /// ```
+  final Future<void> Function(
+    String attributeId,
+    int importanceIndex,
+  ) saveImportance;
+
+  /// Loads the importance of an attribute, e.g. from a SharedPreferences.
+  ///
+  /// Ex:
+  /// ```dart
+  /// (String attributeId) =>
+  ///     mySharedPreferences.getInt(attributeId)
+  ///     ?? PreferenceImportance.INDEX_NOT_IMPORTANT;
+  /// ```
+  final int Function(String attributeId) loadImportance;
+
+  /// 2-letter language code of the latest reference load.
+  String _languageCode;
+
+  /// "was it a https load" bool of the latest reference load.
+  bool _isHttps;
+
+  String get languageCode => _languageCode;
+  bool get isHttps => _isHttps;
+
+  List<PreferenceImportance> _preferenceImportances;
+  Map<String, int> _preferenceImportancesReverse;
+  List<AttributeGroup> _attributeGroups;
+
+  List<AttributeGroup> get attributeGroups => _attributeGroups;
+
+  // TODO(stephanegigandet): move to AttributeGroup?
```

Comment:
Yes, that's a good idea, attributes can be used independently of the product preferences, so those const strings should be in AttributeGroups and Attributes.